### PR TITLE
Fix/issue 513

### LINKED ISSUE #513: gfx/matrix.h: VC++ warning about loss of precision from `double` to `float`, on 9 separate lines in this file
Compiling Vega Strike on Windows generates a whole slew of compiler warnings. Many of them are repeated over and over again, from the same header files included multiple times in different .cpp files.

One such repeated warning mentions a possible loss of precision converting from `double` to `float`, in `engine/src/gfx/matrix.h`. This warning occurs on **nine separate lines** in this file, for a total number of occurrences equal to 9 times however many files matrix.h is included in.

We could cut down the number of compiler warnings a LOT by just fixing this file. And it looks like a simple fix to me: Just change

```C++
    double c = cos( angle );
    double s = sin( angle );
```

in inline function Rotate to

```C++
    float c = cos( angle );
    float s = sin( angle );
```

The overloaded `cos` and `sin` functions return `float`s in this case anyway. Then those `float` values are converted to `double`s, and then back to `float`s. It doesn't make much sense, at least as far as I can see.

### PR DESCRIPTION
**Pull request by stephengtuggy**
Thank you for submitting a pull request and becoming a contributor to the Vega Strike Core Engine.

Please answer the following:

Code Changes:
- [X] Have the PR Validation Tests been run? See https://github.com/vegastrike/Vega-Strike-Engine-Source/wiki/Pull-Request-Validation
- [ ] This is a documentation change only

Issues:
- Please list any related issues
- Closes #513 

Purpose:
- What is this pull request trying to do? Fix a set of compiler warnings
- What release is this for? 0.9.x / 1.0.x
- Is there a project or milestone we should apply this to? 0.9.x / 1.0.x

### CODE REVIEW

**Comment by stephengtuggy** [comment]
I may be imagining it, but it seems like my ship actually rolls more smoothly with this patch.

File: engine/src/gfx/matrix.h [hunk 1]
```diff
@@ -355,8 +360,8 @@ inline void InvertMatrix( Matrix &o, const Matrix &trans )
 
 inline void Rotate( Matrix &tmp, const Vector &axis, float angle )
 {
-    double c = cos( angle );
-    double s = sin( angle );
+    float c = cos( angle );
+    float s = sin( angle );
```
Comment (BenjamenMeyer) [inline engine/src/gfx/matrix.h:364]: Okay, so this means we don't lose precision on the cos/sin of angle.
Question: do this mean we end up losing precision in M() when processing Matrix and Vector data? Is any of that data stored as `double`? or is it it all `float`?

**Review by BenjamenMeyer — COMMENTED** [review]
Just a question before I approve

File: engine/src/gfx/matrix.h (hunk 1 above)
Comment (stephengtuggy) [reply engine/src/gfx/matrix.h:364]: @BenjamenMeyer I just checked, and all of the Matrix and Vector data is stored as `float`.

**Review by BenjamenMeyer — APPROVED** [review]

**Pull request merged**